diff --git a/payment/payment.info b/payment/payment.info
index e8231df..b1df8ce 100644
--- a/payment/payment.info
+++ b/payment/payment.info
@@ -2,9 +2,9 @@ name = Payment
 description = "Allows other modules to let their users make payments using the payment methods connected to Payment."
 core = 7.x
 dependencies[] = entity
-files[] = includes/payment.inc
-files[] = includes/views/PaymentViewsHandlerFieldPaymentMethodControllerTitle.inc
-files[] = includes/views/PaymentViewsHandlerFieldPaymentMethodControllerDescription.inc
-files[] = includes/views/PaymentViewsHandlerFieldPaymentMethodControllerOperations.inc
-files[] = includes/views/PaymentViewsHandlerFieldPaymentStatus.inc
-files[] = includes/views/PaymentViewsHandlerFieldPaymentOperations.inc
\ No newline at end of file
+files[] = payment.classes.inc
+files[] = views/PaymentViewsHandlerFieldPaymentMethodControllerTitle.inc
+files[] = views/PaymentViewsHandlerFieldPaymentMethodControllerDescription.inc
+files[] = views/PaymentViewsHandlerFieldPaymentMethodControllerOperations.inc
+files[] = views/PaymentViewsHandlerFieldPaymentStatus.inc
+files[] = views/PaymentViewsHandlerFieldPaymentOperations.inc
\ No newline at end of file
diff --git a/payment/payment.module b/payment/payment.module
index 5a69ef4..882a4f3 100644
--- a/payment/payment.module
+++ b/payment/payment.module
@@ -198,7 +198,7 @@ function payment_payment_method_controller_info() {
 function payment_views_api() {
   return array(
     'api' => 3,
-    'path' => drupal_get_path('module', 'payment') . '/includes/views',
+    'path' => drupal_get_path('module', 'payment') . '/views',
   );
 }
 
diff --git a/paymentui/.DS_Store b/paymentui/.DS_Store
index 53c8059..da832d2 100644
Binary files a/paymentui/.DS_Store and b/paymentui/.DS_Store differ
diff --git a/paymentui/paymentui.module b/paymentui/paymentui.module
index 5c4aad5..3d26a9e 100644
--- a/paymentui/paymentui.module
+++ b/paymentui/paymentui.module
@@ -12,8 +12,7 @@ function paymentui_menu() {
     'page arguments' => array(1),
     'access callback' => 'paymentui_payment_access',
     'access arguments' => array(1),
-    'file' => 'paymentui.payment.inc',
-    'file path' => drupal_get_path('module', 'paymentui') . '/includes',
+    'file' => 'paymentui.payment_pages.inc',
   );
   $items['payment/%paymentui_menu_payment/view'] = array(
     'title' => 'View',
@@ -33,7 +32,6 @@ function paymentui_menu() {
     'page callback' => 'payment_page_payment_method_add_select_controller',
     'access arguments' => array('paymentui.payment_method.administer'),
     'file' => 'paymentui.admin.inc',
-    'file path' => drupal_get_path('module', 'paymentui') . '/includes',
     'type' => MENU_LOCAL_ACTION,
     'weight' => 1,
   );
@@ -43,7 +41,6 @@ function paymentui_menu() {
     'page arguments' => array(6),
     'access arguments' => array('paymentui.payment_method.administer'),
     'file' => 'paymentui.admin.inc',
-    'file path' => drupal_get_path('module', 'paymentui') . '/includes',
     'type' => MENU_CALLBACK,
   );
   $items['admin/config/services/payment/method/%paymentui_menu_payment_method'] = array(
@@ -53,7 +50,6 @@ function paymentui_menu() {
     'title callback' => 'paymentui_payment_method_title',
     'title arguments' => array(5),
     'file' => 'paymentui.admin.inc',
-    'file path' => drupal_get_path('module', 'paymentui') . '/includes',
   );
   $items['admin/config/services/payment/method/%paymentui_menu_payment_method/edit'] = array(
     'title' => 'Edit',
@@ -66,7 +62,6 @@ function paymentui_menu() {
     'access arguments' => array('paymentui.payment_method.administer'),
     'title' => 'Delete',
     'file' => 'paymentui.admin.inc',
-    'file path' => drupal_get_path('module', 'paymentui') . '/includes',
     'type' => MENU_LOCAL_TASK,
     'context' => MENU_CONTEXT_INLINE,
     'weight' => 1,
@@ -79,7 +74,6 @@ function paymentui_menu() {
     'page arguments' => array('paymentui_form_global_configuration'),
     'access arguments' => array('paymentui.payment_method.administer'),
     'file' => 'paymentui.admin.inc',
-    'file path' => drupal_get_path('module', 'paymentui') . '/includes',
     'type' => MENU_LOCAL_TASK,
     'weight' => 2,
   );
@@ -150,7 +144,7 @@ function paymentui_form_alter(array &$form, array &$form_state, $form_id) {
 function paymentui_views_api() {
   return array(
     'api' => 3,
-    'path' => drupal_get_path('module', 'paymentui') . '/includes/views',
+    'path' => drupal_get_path('module', 'paymentui') . '/views',
   );
 }
 
